### PR TITLE
Add GLMPPSeq: PPSeq with GLM Background Model

### PR DESCRIPTION
**Pull request by smejak**
### Summary

This PR introduces `GLMPPSeq`, a generalized version of `PPSeq` that incorporates a Generalized Linear Model (GLM) for modeling time-varying background firing rates. This extends the original PPSeq framework to handle non-stationary neural data where background activity varies over time.

### Key Features

**GLMPPSeq Model:**
- **Time-varying background**: Models background firing rates using a GLM with RBF (Radial Basis Function) covariates, replacing the constant background rate in vanilla PPSeq
- **Flexible covariate system**: Uses smooth temporal basis functions to capture slow changes in population activity
- **Sparsity regularization**: Supports L1 penalties on both sequence amplitudes (`l1_amp`) and template neuron participation (`l1`)
- **Configurable RBF basis**: `rbf_width` parameter controls temporal smoothness of background (scale factor relative to basis spacing, default=0.2)
- **Efficient Newton-Raphson updates**: Vectorized GLM parameter optimization with numerical stability safeguards

### Implementation Details

- EM algorithm with three update steps:
  1. **E-step**: Update sequence amplitudes given current parameters
  2. **M-step (background)**: Update GLM coefficients via Newton-Raphson
  3. **M-step (templates)**: Update Gaussian template parameters via moment matching

- Proper GLM initialization in `initialize_random()` and `initialize_default()` 
- Consistent use of `phi` for covariates throughout the class
- Device-aware tensor operations for GPU acceleration

### Backward Compatibility

- `PPSeq` and `CAVI` classes remain unchanged
- New model is in separate `GLMPPSeq` class to avoid breaking existing code

### Testing Recommendations

After merge, users should verify:
- Model initializes without errors across different parameter settings
- `fit()` runs successfully on both CPU and GPU
- GLM background rates properly capture temporal dynamics
- L1 regularization produces expected sparsity

### CODE REVIEW

File: ppseq/model.py [hunk 1]
```diff
@@ -11,6 +8,473 @@
...
+
+            # Clip gradients to prevent explosion
+            delta = torch.clamp(delta, min=-1.0, max=1.0)
+
+            # Check for NaN/Inf before updating
+            if torch.isnan(delta).any() or torch.isinf(delta).any():
+                print("Warning: NaN or Inf detected in GLM update, stopping iteration")
+                break
+
+            self.beta = self.beta + 0.1 * delta
```
Comment (xanderladd) [inline ppseq/model.py:277]: we should allow this 0.1 to configured - sets the learning rate for newton updates

File: ppseq/model.py [hunk 2]
```diff
@@ -11,6 +8,473 @@
...
+        """
+        Vectorized Newton-Raphson update for GLM parameters
+        Updates all neurons simultaneously for speed
+        """
+        D = self.template_duration
+        T = data.shape[1]
+        N = self.num_neurons
+
+        # Store T for RBF basis functions
+        self.T_for_rbf = T
```
Comment (xanderladd) [inline ppseq/model.py:237]: If there is a better way to do T for RBF, it would be nice - if you want to do inference, it needs to be set again and can cause some issues... though model isn't currently set for inference. but we can maybe implement this in a cleaner way next time

**Review by xanderladd — APPROVED** [review]
this looks good.

a few things to consider going forward:
- are we clipping `eta` (line 129) and `delta` (line 270) to agressively? to leinent? we can  set these thresholds empirically
- same as above for `scales = torch.relu(scales - self.l1)` 
- if we add inference mode T_for_RBF will be annoying - RBF basis wont have correct time axis
- do we ever want to add RBF basis with different lengthscale / width
